### ISSUE TITLE
SSD keep_difficult=False Problem?

### ISSUE THREAD

**Issue by shwoo93** [amdegroot/ssd.pytorch#54]
It seems the code trains the network(SSD) without difficult training sets and evaluates with out difficult test sets. This seems unfair to compare the final mAP of this code with the original caffe code which uses difficult sets for both trianing and testing.

Additionally I also trained the network with 07 ++ 12 train set (07 trainval, 07 test, 12 trainval) and tested with 12 test set using the official server. And the result was 74.1%, 2% below from the latest version of SSD300(75.8%). Since the server's ground truth would contain difficult sets, but the network which was trained only with easy sets would not be able to achieve the original performance.
